nixos/modules/system/boot/loader/grub/grub.nix

```diff
@@ -769,6 +769,16 @@ in
         '';
       };
 
+      loadDeviceTree = mkOption {
```

Comment:
I suggest instead renaming the existing `boot.loader.systemd-boot.installDeviceTree` to `boot.loader.efi.installDeviceTree` and reusing that here (as done in the first commit of https://github.com/NixOS/nixpkgs/pull/439700).